qualtran/bloqs/reflection_using_prepare.py

```diff
@@ -63,6 +65,7 @@ class ReflectionUsingPrepare(GateWithRegisters):
     """
 
     prepare_gate: PrepareOracle
+    extra_registers: Tuple[Tuple[str, int], ...] = ()
```

Comment:
Maybe directly expect a `Tuple[Register, ...]` instead of ` Tuple[Tuple[str, int], ...]` ?

Also, add the `extra_registers` as the last attribute in the after `global_phase` so existing call sites that pass `*args` instead of `**kwargs` are not broken